File: src/main/java/std/nooook/readinggardenkotlin/common/config/TaskSchedulerConfig.java
package std.nooook.readinggardenkotlin.common.config;

import java.time.Clock;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.TaskScheduler;
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;

@Configuration(proxyBeanMethods = false)
public class TaskSchedulerConfig {

    @Bean
    public TaskScheduler taskScheduler() {
        ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
        scheduler.setPoolSize(1);
        scheduler.setThreadNamePrefix("reading-garden-auth-");
        scheduler.initialize();
        return scheduler;
    }

    @Bean
    public Clock utcClock() {
        return Clock.systemUTC();
    }
}

File: src/main/kotlin/std/nooook/readinggardenkotlin/common/config/TaskSchedulerConfig.kt
package std.nooook.readinggardenkotlin.common.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.TaskScheduler
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler
import java.time.Clock

@Configuration
class TaskSchedulerConfig {
    @Bean
    fun taskScheduler(): TaskScheduler =
        ThreadPoolTaskScheduler().apply {
            poolSize = 1
            setThreadNamePrefix("reading-garden-auth-")
            initialize()
        }

    @Bean
    fun utcClock(): Clock = Clock.systemUTC()
}
